### PR TITLE
Fix non-routing Link component usages (#210)

### PR DESCRIPTION
**Pull request by denis-sokolov**
#210

### CODE REVIEW

File: src/links.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import {
+  Link as ReactRouterLink,
+  NavLink as ReactRouterNavLink,
+} from 'react-router-dom'
+
+function isRoutingUrl (to) {
+  if (typeof to !== 'string') return true
+  return !to.match(/^#/) && !to.match(/^[a-z]{1,10}:\/\//)
```
Comment (EmilTholin) [inline src/links.js]: Will this work when using the [`hash` version of the Router](https://github.com/nozzle/react-static#router)?

Comment (denis-sokolov) [reply src/links.js]: Good point, I haven’t tried it.

It should work, as the user of the components should still provide a proper, non-fragment URL, it is the responsibility of `history`/`react-router` to encode the output in whatever format they currently need.

Comment (denis-sokolov) [reply src/links.js]: I can try and verify it, maybe tomorrow.

**Comment by EmilTholin** [comment]
The [documentation for `Link` and `NavLink`](https://github.com/nozzle/react-static#link-and-navlink) should probably be updated as well to indicate that they are custom `react-static` components now, since `react-router` [doesn't support hash links](https://github.com/ReactTraining/react-router/issues/394#issuecomment-252511289).

It might also cause some confusion that a hash `NavLink` will not have an `active` class or `aria-current` attribute added to it when it is active.

File: src/links.js (hunk 1 above)
Comment (denis-sokolov) [reply src/links.js]: No meaningful BC breakages, here are the findings:

1. `<Router type="hash">` and regular links keep working. `<Link to="about">` matches our `isRoutingUrl` check and thus gets passed through to react-router, which renders a `<a href="#/about">`.

2. `<Router type="hash">` and links that manually have the hash route kind of work. `<Link to="#/about">` does not match our `isRoutingUrl` check and gets rendered with the newly introduced shortcut into a `<a href="#/about">`. react-router handles this links, so it responds to user interaction and changes routes, but it does not have `aria-current`. We could attempt to distinguish `to="#/about"` as a non-fragment link, but technically ids [can start with a slash](https://www.w3.org/TR/2011/WD-html5-20110525/elements.html#the-id-attribute). The usage `<Link to="#/about">` is already kind of wrong, and with proper history support [nearly universal](https://caniuse.com/#feat=history), I don’t think we have to worry about this.

3. `<Router type="hash">` and fragment links are still broken after the latest changes. `<Link to="#subheading">` gets rendered with the newly introduced shortcut into a `<a href="#subheading">`, but is still handled by react-router to navigate to `#/subheading`, ignoring the fragment link. This [was already broken](https://github.com/ReactTraining/react-router/issues/394) with the react-router `<Link>`, and is still broken with our manually rendered `<a>`. Nothing to see here.

**Comment by denis-sokolov** [comment]
> The documentation for Link and NavLink should probably be updated as well to indicate that they are custom react-static components now, since react-router doesn't support hash links.

If a package user decides that for hash links they need a manual `<a>`, good for them. They control their fragment links, I am happy for them. This change was meant to be invisible, to stop counter-intuitive behavior of a broken `<Link>`. It’s not meant to canonically replace all `<a>` in users’ applications. Thus, I don’t think it warrants a mention in the documentation.

> It might also cause some confusion that a hash NavLink will not have an active class or aria-current attribute added to it when it is active.

Hypothetically, yes. But those are extra features that someone with an understanding of routing should expect, but then they should also expect them to not work. Still, you’re right, so I’ve added a warning in #232.

**Pull request merged**